nemo_skills/dataset/deep-swe/prepare.py

```
@@ -0,0 +1,267 @@
+
+def _as_dict(value):
+    if value is None:
+        return {}
+    if hasattr(value, "unwrap"):
+        return value.unwrap()
+    return dict(value)
+
+
+def _clone_or_update_repo(repo_url: str, dest: Path, commit: str | None) -> Path:
+    if dest.exists() and (dest / ".git").exists():
+        subprocess.check_call(["git", "-C", str(dest), "fetch", "--depth", "1", "origin"], cwd=dest)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)